chocolatey/docs
# (#798) Add AnchorJS anchors to list items

## Description Of Changes

This adds anchors to list items that contain a `<strong>` or a `<code>`
node directly after an opening `<li>` tag. This is to make it easier to
navigate to these items, as often times this format is used more as a
"definition", or something we'd want to potentially link to.

No changes to current markdown is needed, as this intended to just
"work". All changes for this functionality live in choco-theme and will
be documented in the Choco Design System.

## Motivation and Context

This feature came to mind while putting together the new Jargon page. It would be nice to be able to link others to specific list items that serve more as "definitions".

## Testing


1. Run this PR
2. Go to http://localhost:5080/en-us/configuration
3. Hover over the list items that contain code as the first item. Notice that a blue "link" icon appears (like it does in headers). Clicking the icon adds the text that inside of the `<code>` onto the url, so that it can be navigated to.
4. On the same page, scroll down to where there are checkboxes in front of the first `<code>` item. The functionality should work the same as in the step above.
5. Now go to http://localhost:5080/en-us/getting-started
6. Hover over any list items that contain a `<strong>` tag as the first item in the list. The functionality should be the same as in step 3.
7. In general, only list items that begin with a `<code>` tag or a `<strong>` tag get anchor links, and the anchor link should contain the text inside of said tag (not the entire contents of the list item itself).

## Change Types Made

## Change Checklist


## Related Issue
* https://github.com/chocolatey/choco-theme/issues/347
* https://github.com/chocolatey/docs/issues/798
* https://github.com/chocolatey/boxstarter.org/issues/61
* https://github.com/chocolatey/choco-design-system/issues/31
* https://github.com/chocolatey/blog/issues/230

> **This issue and tags are synced with choco-theme and will be closed in choco-theme once all related issues have been resolved. Please do not close the issue or update tags on this repository.**
